lib/vtls/openssl.c

```diff
@@ -2987,6 +3043,15 @@ static CURLcode ossl_connect_step1(struct connectdata *conn, int sockindex)
       infof(data, "error importing windows ca store, continuing anyway\n");
   }
 #endif
+  if(ca_file_pem && load_cacert_from_memory(backend->ctx, ca_file_pem)) {
+    if(verifypeer) {
+      /* Fail if we insist on successfully verifying the server. */
+      failf(data, "error setting certificate from memory");
+      return CURLE_SSL_CACERT_BADFILE;
+    }
+    /* Continue with a warning if no certificate verification is required. */
+    infof(data, "error setting certificate file, continuing anyway\n");
```

Comment:
Should we really ignore the failure to load the cacert from memory when we are not verifying the peer certificate?

Would not it be better to fail anyway? Honest question, I don't have a specific opinion on this matter.